validation-test/compiler_crashers_2_fixed/rdar74557857.swift

```diff
@@ -0,0 +1,7 @@
+// RUN: %target-swift-frontend -typecheck %s
+
+extension Optional {
+  init?() {
```

Comment:
I'm afraid this needs more tests because replaceCovariantResultType() can't distinguish the Optional type in the failable initializer return from a bona-fide initializer on Optional.

I would try to test the following with SILGen tests:
- Actually calling the failable init on Optional
- returning nil from the failable init on Optional